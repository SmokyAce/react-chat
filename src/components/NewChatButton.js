import React from 'react';
import { withStyles } from 'material-ui/styles';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogTitle
} from 'material-ui/Dialog';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import AddIcon from 'material-ui-icons/Add';

const styles = theme => ({
  newChatButton: {
    position: 'absolute',
    left: 'auto',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3 + 48 // + bottom navigation
  },
  dialog: {
    minWidth: '200px'
  }
});

class NewChatButton extends React.Component {
  state = {
    newChat: {
      value: '',
      isValid: true
    },
    open: false
  };

  handleInputChange = event => {
    this.setState({
      newChat: {
        value: event.target.value,
        isValid: true
      }
    });
  };

  handleCreate = () => {
    const { newChat } = this.state;
    if (newChat.value === '') {
      return this.setState(prevState => ({
        ...prevState,
        newChat: {
          value: '',
          isValid: false
        }
      }));
    }

    this.props.createChat(newChat.value).then(() =>
      this.setState({
        open: false,
        newChat: {
          value: '',
          isValid: true
        }
      })
    );
  };

  handleNewChat = () => {
    this.setState(prevState => ({
      ...prevState,
      open: !prevState.open
    }));
  };

  render() {
    const { classes } = this.props;
    const { open, newChat } = this.state;

    return (
      <React.Fragment>
        <Button
          variant='fab'
          color='primary'
          className={classes.newChatButton}
          onClick={this.handleNewChat}
        >
          <AddIcon />
        </Button>
        <Dialog
          maxWidth='xs'
          fullWidth
          open={open}
          onClose={this.handleNewChat}
        >
          <DialogTitle>Create new chat</DialogTitle>
          <DialogContent>
            <TextField
              fullWidth
              required
              label='New chat'
              placeholder='Type chat name ...'
              name='newChat'
              value={newChat.value}
              error={!newChat.isValid}
              onChange={this.handleInputChange}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleCreate} color='primary'>
              Create
            </Button>
          </DialogActions>
        </Dialog>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(NewChatButton);
