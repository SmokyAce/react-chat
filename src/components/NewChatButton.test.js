/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewChatButton from './NewChatButton';

const mockProps = {
  disabled: false,
  createChat: jest.fn(),
};

describe('<NewChatButton />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewChatButton {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<NewChatButton {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly disabled', () => {
    const tree = renderer.create(<NewChatButton {...mockProps} disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
