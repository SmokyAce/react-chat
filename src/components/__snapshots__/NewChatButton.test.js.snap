// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewChatButton /> renders correctly 1`] = `
<button
  className="MuiButtonBase-root-60 MuiFab-root-50 MuiFab-primary-52 NewChatButton-newChatButton-48"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiFab-label-51"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-63"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-88"
  />
</button>
`;

exports[`<NewChatButton /> renders correctly disabled 1`] = `
<button
  className="MuiButtonBase-root-60 MuiButtonBase-disabled-61 MuiFab-root-50 MuiFab-primary-52 MuiFab-disabled-56 NewChatButton-newChatButton-48"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiFab-label-51"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-63"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</button>
`;
